README.md

```diff
@@ -295,16 +398,380 @@ Here is an example:
+JSON object:
+
+``` emacs-lisp
+(setq json-people "{
+  \"people\": [
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"John\",
+          \"last\": \"Johnson\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"CEO\",
+        \"location\": {
+          \"country\": \"Norway\",
+          \"city\": \"Oslo\"
+        }
+      },
+      \"skill-refs\": [
+        \"skill-1\",
+        \"skill-3\" 
+      ]
+    },
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"Sammy\",
+          \"last\": \"Smith\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"VP, Marketing\",
+        \"location\": {
+          \"country\": \"USA\",
+          \"city\": \"Peoria\"
+        }
+      },
+      \"skill-refs\": [
+        \"skill-2\",
+        \"skill-3\" 
+      ]
+    },
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"Tommy\",
+          \"last\": \"Twiddler\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"Mail Distribution\",
+        \"location\": {
+          \"country\": \"USA\",
+          \"city\": \"Peoria\"
+        }
+      }
+    }
+  ],
+  \"skills\": {
+    \"skill-1\": {
+      \"name\": \"Programming\",
+      \"description\": \"Can code like Richard S or Linus T\"
+    },
+    \"skill-2\": {
+      \"name\": \"Design\",
+      \"description\": \"Can think out loud, structure code systems\"
+    },
+    \"skill-3\": {
+      \"name\": \"Communication\",
+      \"description\": \"Can explain complicated things to social scientists\"
+    }
+  }
+}")
+```
+
+When you execute the code below:
+
+``` emacs-lisp
+(let* ((people-data (jeison-read jeison-people json-people)))
+  (mapcar (lambda (person)
+            (princ (format "%s skills: %s\n"
+                           (oref person name)
+                           (if (oref person skill-refs)
+                               ; The person has skill references
+                               (mapconcat (lambda (the-skill-ref)
+                                            (let ((the-skill (gethash (intern the-skill-ref)
+                                                                      (oref people-data skills))))
+                                              (oref the-skill name)))
+                                          (oref person skill-refs)
+                                          ", ")
+                               ; No skill references for this person...
+                               "Missing or unknown skills"))))
+          (oref people-data people)))
+```
+
+You should see this output in *ielm*:
+
+```
+Johnson skills: Programming, Communication
+Smith skills: Design, Communication
+Twiddler skills: Missing or unknown skills
+```
```

Comment:
This is a really huge example for a very simple and straightforward feature. I also expect users not to read this file end-to-end (hash tables is quite an unusual use case), so adding dependency on a less essential feature is not helping. I suggest to decouple them and drastically cut the example here.